src/distributing-policies/secure-supply-chain.md

```diff
@@ -0,0 +1,387 @@
+# Secure Supply Chain
+
+Users can configure Kubewarden to run policies signed by only entities that they
+trust by using the [sigstore](https://www.sigstore.dev) project. This means that
+policy developers can sign their policies and publish them in a registry.
+The cluster operators can run these signed policies and ensure the validity of the policies chain of custody
+
+
+## Signing the policies
+
+Kubewarden recommends using the [cosign](https://github.com/sigstore/cosign) project
+for signing policies. This subsection details a key-based method of signing policies
+in which, users need to generate a private-public key pair. The generated keys help
+in verifying whether the signed artifacts came from the expected user i.e. the
+owner of the keys. Using the `cosign generate-key-pair` command, as illustrated
+below, it is possible to generate the aforementioned keypair.
+
+```bash
+cosign generate-key-pair
+Enter password for private key:
+Enter password for private key again:
+Private key written to cosign.key
+Public key written to cosign.pub
+```
+
+Once the keys are generated, users can use it to sign policies.
+
+> **WARNING**: The private key file,  `cosign.key`, should not be shared. This
+> is a secret file that should be used only by the owner of the key for signing policies.
+
+To sign a policy users can execute the `cosign sign` command passing the `--key`
+command line argument with their private key file:
+
+```bash
+cosign sign --key cosign.key ghcr.io/jvanz/policies/user-group-psp:latest
+an error occurred: no provider found for that key reference, will try to load key from disk...
+Enter password for private key:
+Pushing signature to: ghcr.io/jvanz/policies/user-group-psp
+```
+
+This command will sign the policy and publish the signature data in the registry
+as well. Now the policies are ready to be used in a Kubewarden installation with signature
+verification enabled.
+
+
+### Keyless signing
+
+Many times the policies are automatically built in CI/CD pipelines which complicates
```

Comment:
I would rephrase to something like:

> Many times the policies are automatically built in CI/CD pipelines which complicates
the key generation process. The Sigstore workflow is specifically designed for these situations and others where key generation is not possible.
Instead of using long-lived singing keys, the keyless workflow uses CAs and certificate chains: a short-lived certificate key is generated, and linked into a chain of trust by completing an identity challenge to confirm the signer's identity. The short-lived certificate key is around just enough for the signing to occur. The identity challenge is performed by authenticating against an OpenID Connect (OIDC) provider, Sigstore's [Fulcio](https://github.com/sigstore/fulcio) service is used for the chain of trust.
> 
> For signing, we will again use Sigstore's `cosign` utility. However, as of this writing, this feature is not enabled by default in `cosign`. So, it's necessary to enable this experimental feature before proceeding with keyless signing as shown below: